File: csharp/Platform.Numbers.Tests/ArithmeticExtensionsTests.cs
using Xunit;

namespace Platform.Numbers.Tests
{
    /// <summary>
    /// <para>
    /// Represents the arithmetic extensions tests.
    /// </para>
    /// <para></para>
    /// </summary>
    public static class ArithmeticExtensionsTests
    {
        /// <summary>
        /// <para>
        /// Tests that increment test.
        /// </para>
        /// <para></para>
        /// </summary>
        [Fact]
        public static void IncrementTest()
        {
            var number = 0UL;
            var returnValue = number.Increment();
            Assert.Equal(1UL, returnValue);
            Assert.Equal(1UL, number);
        }

        /// <summary>
        /// <para>
        /// Tests that decrement test.
        /// </para>
        /// <para></para>
        /// </summary>
        [Fact]
        public static void DecrementTest()
        {
            var number = 1UL;
            var returnValue = number.Decrement();
            Assert.Equal(0UL, returnValue);
            Assert.Equal(0UL, number);
        }
    }
}

File: cpp/Platform.Numbers.Tests/ArithmeticExtensionsTests.cpp
namespace Platform::Numbers::Tests
{
    TEST_CLASS(ArithmeticExtensionsTests)
    {
        public: TEST_METHOD(IncrementTest)
        {
            auto number = 0UL;
            auto returnValue = number.Increment();
            Assert::AreEqual(1UL, returnValue);
            Assert::AreEqual(1UL, number);
        }

        public: TEST_METHOD(DecrementTest)
        {
            auto number = 1UL;
            auto returnValue = number.Decrement();
            Assert::AreEqual(0UL, returnValue);
            Assert::AreEqual(0UL, number);
        }
    };
}
